pkg/actions/lua/lakefs/catalogexport/delta_exporter.lua

```diff
@@ -0,0 +1,134 @@
+local lakefs = require("lakefs")
+local formats = require("formats")
+local pathlib = require("path")
+local json = require("encoding/json")
+local utils = require("lakefs/catalogexport/internal")
+
+local function delta_log_entry_key_generator()
+    local current = 0
+    return function()
+        local delta_log_entry_length = 20
+        local key = tostring(current)
+        local padding_length = delta_log_entry_length - key:len()
+        local padded_key = ""
+        for _ = 1, padding_length do
+            padded_key = padded_key .. "0"
+        end
+        padded_key = padded_key .. key
+        current = current + 1
+        return padded_key
+    end
+end
+
+local function get_delta_client(key, secret, region)
+    return formats.delta_client("s3", key, secret, region)
+end
+
+--[[
+    action:
+        - repository_id
+        - commit_id
+
+   table_paths: ["path/to/table1", "path/to/table2", ...]
+
+    storage_client:
+        - put_object: function(bucket, key, data)
+
+    delta_client:
+        - get_table: function(repo, ref, prefix)
+
+]]
+local function export_delta_log(action, table_paths, storage_client, delta)
+    local repo = action.repository_id
+    local commit_id = action.commit_id
+
+    local ns = utils.get_storage_namespace(repo)
+    if ns == nil then
+        error("failed getting storage namespace for repo " .. repo)
```

Comment:
There's no need calling lakeFS for that, you can get the storage namespace from the `action` global, it has fields [storage_namespace](https://github.com/treeverse/lakeFS/blob/e5ca6b86baa2c38d2c3ddaaca0db3b014d15906f/pkg/actions/lua.go#L55) see usage in [glue_exporter.lua](https://github.com/treeverse/lakeFS/blob/e5ca6b86baa2c38d2c3ddaaca0db3b014d15906f/pkg/actions/lua/lakefs/catalogexport/glue_exporter.lua#L116).

Reply:
very cool
didn't notice that one